src/edu/agh/wfiis/designpatterns/patternsUsageExample/patterns/DesignPattern.java

```diff
@@ -0,0 +1,77 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.patterns;
+
+import edu.agh.wfiis.designpatterns.patternsUsageExample.enums.DesignPatternType;
+
+public class DesignPattern {
+
+    private String name;
+    private String patternCode;
+    private String description;
+    private String example;
+
+    private DesignPatternType type;
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getExample() {
+        return example;
+    }
+
+    public String getPatternCode() {
+        return patternCode;
+    }
+
+    public DesignPatternType getType() {
+        return type;
+    }
+
+    public boolean isTheSameTypeAs(DesignPattern patternToCompare) {
+        return type == patternToCompare.getType();
+    }
+
+    private DesignPattern(DesignPattern.Builder builder) {
+        this.name = builder.name;
+        this.description = builder.description;
+        this.example = builder.example;
+        this.patternCode = builder.patternCode;
+        this.type = builder.type;
+    }
+
+    public static class Builder {
+        private String name;
```

Comment:
Dobre zastosowanie Buildera do tworzenia obiektu DesignPattern.
Skoro w wymaganych parametrach są dwa Stringi, których kolejność można pomylić, warto rozważyć zaimplementowanie interfejsów, które wymuszą nie tylko podanie wymaganych wartości, ale także ich kolejność, zgodnie z przykładem: 
https://aidium.se/2015/01/builder-pattern-for-mandatory-values/